Orange/widgets/evaluate/owtestlearners.py

```diff
@@ -443,13 +443,14 @@ def _update_results(self):
         items = [(key, slot) for key, slot in self.learners.items()
                  if slot.results is None]
         learners = [slot.learner for _, slot in items]
-        if len(items) == 0:
-            return
 
         if self.test_data is not None and \
                 self.resampling != OWTestLearners.TestOnTest:
             self.Warning.test_data_unused()
 
+        if len(items) == 0:
```

Comment:
Instead of moving the check on `items` away from its creation, how about moving the check for `TestOnTest` up, together with other similar checks. The `if`s can even be merged.

Otherwise this seems to fix the problem :)